File: platines-webapp/platines-back/src/main/java/fr/asipsante/platines/entity/enums/Role.java
/*
 * (c) Copyright 2017-2024, ANS. All rights reserved.
 */
package fr.asipsante.platines.entity.enums;

/**
 * The Enum Role.
 *
 * @author apierre
 */
public enum Role {

  /** Application client. */
  CLIENT("Client"),
  /** Application server. */
  SERVER("Serveur");

  /** value. */
  private String value;

  /**
   * constructeur.
   *
   * @param value
   */
  private Role(String value) {
    this.value = value;
  }

  /**
   * @return the value
   */
  public String getValue() {
    return value;
  }
}

File: platines-webapp/platines-front/src/app/pages/entity/role.ts
///
/// (c) Copyright 2017-2024, ANS. All rights reserved.
///

export const CLIENT: string='CLIENT';
export const SERVER: string='SERVER';

export function toTestWebserviceRole(applicationRole: string): string {
    switch(applicationRole){
        case CLIENT:
            return SERVER;
        case SERVER:
            return CLIENT;
        default:
            throw `Unsupported role ${applicationRole}`
    }
}